integration-tests/bats/sql-branch.bats

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+
+    dolt sql <<SQL
+CREATE TABLE test (
+    pk int primary key
+);
+
+INSERT INTO test VALUES (0),(1),(2);
+SQL
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
+@test "sql-branch: DOLT_BRANCH works" {
+    run dolt branch
+    [[ ! "$output" =~ "new_branch" ]] || false
+
+    run dolt sql -q "SELECT DOLT_BRANCH('new-branch')"
+    [ $status -eq 0 ]
+
+    # It changes the branch for that session which ends after the SQL
+    # statements are executed.
```

Comment:
I don't understand this comment. Are you saying that the SQL session for line 25 has its branch changed? 

`git branch abc` and `dolt branch abc` both create a new branch, but do not switch to it.